ci/functional/test_main_prep_node.sh

```diff
@@ -2,27 +2,62 @@
-            ifup "$iface" || true
-            ifconfig -a || true
-            ls -l /etc/sysconfig/network-scripts/ifcfg-* || true
-            cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
-          } 2>&1 | mail -s "Interface $iface found down after reboot" \
-                   -r "$HOSTNAME"@intel.com "$OPERATIONS_EMAIL"
+            mkdir -p /tmp/artifacts
+            systemctl status > /tmp/artifacts/systemctl\ status || true
+            systemctl --failed > /tmp/artifacts/systemctl\ --failed || true
+            journalctl -n 500 > /tmp/artifacts/journalctl\ -n\ 500 || true
+            dmesg > /tmp/artifacts/dmesg || true
+            ifconfig "$iface" > /tmp/artifacts/ifconfig\ "$iface" || true
+            cat /sys/class/net/"$iface"/mode > \
+                /tmp/artifacts/cat\ _sys_class_net_"${iface}"_mode || true
+            ifup "$iface" > /tmp/artifacts/ifup\ "$iface" || true
+            ifconfig -a > /tmp/artifacts/ifconfig\ -a || true
+            ls -l /etc/sysconfig/network-scripts/ifcfg-* > \
+                /tmp/artifacts/ls\ -l\ _etc_sysconfig_network-scripts_ifcfg-\* || true
+            cat /etc/sysconfig/network-scripts/ifcfg-"$iface" > \
+                /tmp/artifacts/cat\ _etc_sysconfig_network-scripts_ifcfg-"$iface" || true
+            send_mail "Interface $iface found down after reboot"  \
+                      "Found interface $iface down after reboot."
         fi
         if ! ifconfig "$iface" | grep "inet "; then
-            echo "Failed to bring up interface $iface on $HOSTNAME. " \
-            "Please file a CORCI ticket."
+            report_junit "Failed to bring up interface $iface on $HOSTNAME after it was found down. 
+Please file a CORCI ticket." "IB_interface_down" "Interface $iface on $HOSTNAME is down and won't come up. 
+Please see ${STAGE_NAME}/framework/ in the artifacts for more info."
+            exit 1
+        fi
+        # make sure verbs driver is present on all IB interfaces
+        mkdir -p /tmp/artifacts
+        if ! fi_info -d "$iface" -l | tee /tmp/artifacts/fi_info\ -d\ "$iface"\ -l |
+          grep verbs; then
+            [ -f /tmp/artifacts/dmesg ] || dmesg > /tmp/artifacts/dmesg || true
+            mst status -v > /tmp/artifacts/mst\ status\ -v || true
+            ibv_devinfo > /tmp/artifacts/ibv_devinfo || true
+            send_mail "Interface $iface found without verbs after reboot"         \
+                      "Found interface $iface without verbs driver after reboot."
+            report_junit "Interface $iface on $HOSTNAME is missing the verbs driver but we do no know how to rectify this so all we can do is fail this test run. 
+Please file a CORCI ticket if you have any suggestsions on how to rectify this state in order to prevent failed test runs." \
+                         "verbs_driver_test"                                                                                \
+                         "Missing verbs driver on $HOSTNAME. 
```

Comment:
(style)  trailing whitespace

Reply:
Intentional.  It's inside a multiline string.